docs/snapshotting/versioning.md

```diff
@@ -0,0 +1,98 @@
+# Firecracker snapshot versioning
+
+This document describes how Firecracker persists its state across multiple versions, diving deep into the snapshot format, encoding, compatibility and limitations. 
+
+## Introduction
+The design behind the snapshot implementation enables version tolerant save and restore across multiple Firecracker versions which we call a version space. For example, one can pause a microVM, save it to disk with Firecracker version **0.26.0** and later load it in Firecracker version **0.24.0**. It also works  in reverse: Firecracker version **0.26.0** loads what  **0.24.0** saves. 
+
+Below is an example graph showing backward and forward snapshot compatibility for a version space formally represented as `[v2; vn]`. This is the general picture, but keep in mind that future corner cases might remove edges from this graph.
+
+![Version graph](
+../images/version_graph.png?raw=true
+"Version graph")
+
+## Overview
+Firecracker persists the microVM state as 2 separate objects:
+ - a **guest memory** file
+ - a **microVM state** file.
+ 
+*The block devices attached to the microVM are not considered part of the state and need to be managed separately.*
+
+### Guest memory
+The guest memory file is agnostic to Firecracker's version tolerant save/restore implementation as its contents are fully owned by the guest OS. The microVM memory is by choice saved as a dump of each page or as a sparse file of guest modified pages only. 
+### MicroVM state
+The microVM state file stores the internal state of the VMM (device emulation, KVM and vCPUs) with 2 exceptions - serial emulation and vsock backend. 
+
+While we continuously improve and extend Firecracker's features by adding new capabilities, devices or enhancements, the microVM state file may change both structurally and semantically with each new release. The state file includes versioning information and each Firecracker release implements distinct save/load logic for the supported version space. 
+
+## MicroVM state file format
+
+###  A closer look
+A microVM state file is further split into four different fields:
+
+| Field | Bits| Description |
+|----|----|----|
+| magic_id | 64 | Firecracker snapshot, architecture (x86_64/aarch64) and storage version.
+| version  | 64 | The snapshot version number internally mapped 1:1 to a specific Firecracker version.
```

Comment:
```suggestion
| version  | 64 | The snapshot version number internally 1:1 mapped to a specific Firecracker version.
```

Also, shouldn't the bits number be 16 here?

Reply:
Fixed.